### PR TITLE
[CN-312] add member statuses for detecting ForceStart scenario

### PR DESCRIPTION
**Pull request by SeriyBg**
For detecting Force Restart we need to store members' state information.
Example of failed members' state:
```yaml
status:
    hazelcastClusterStatus:
      readyMembers: 0/2
    members:
    - connected: false
      message: back-off 5m0s restarting failed container=hazelcast pod=hazelcast-0_default(049b9af9-e458-4504-a184-54e22b23134f)
      reason: CrashLoopBackOff
    message: 'multiple (2) errors: pod hazelcast-0 in namespace default failed for
      back-off 5m0s restarting failed container=hazelcast pod=hazelcast-0_default(049b9af9-e458-4504-a184-54e22b23134f):
      CrashLoopBackOff'
    phase: Failed
```

Example of ready members:
```yaml
  status:
    hazelcastClusterStatus:
      readyMembers: 3/3
    members:
    - connected: true
      ip: 10.124.2.56
      podName: ""
      uid: fe00f86d-43a1-4996-905b-6a74131f2b73
      version: 5.0.2
    - connected: true
      ip: 10.124.1.37
      podName: ""
      uid: e01c94d3-e49d-4681-9b8a-1c63913b2207
      version: 5.0.2
    - connected: true
      ip: 10.124.0.40
      podName: ""
      uid: e72079ee-acd4-46e0-bc08-1f9bd011cb0e
      version: 5.0.2
    phase: Running
```

### CODE REVIEW

File: controllers/hazelcast/hazelcast_status.go [hunk 1]
```diff
@@ -56,12 +61,53 @@ func (o optionsBuilder) withExternalAddresses(addrs string) optionsBuilder {
...
 }
 
+func statusMembers(m map[string]cluster.MemberInfo) []hazelcastv1alpha1.HazelcastMemberStatus {
+	members := make([]hazelcastv1alpha1.HazelcastMemberStatus, 0, len(m))
+	for uid, member := range m {
+		a := member.Address.String()
+		members = append(members, hazelcastv1alpha1.HazelcastMemberStatus{
+			Uid:     uid,
+			Ip:      a[:strings.IndexByte(a, ':')],
+			Version: fmt.Sprintf("%d.%d.%d", member.Version.Major, member.Version.Minor, member.Version.Patch),
```
Comment (mtyazici) [inline controllers/hazelcast/hazelcast_status.go:71]: I think, we should add podName here as well.

File: controllers/hazelcast/hazelcast_status.go [hunk 2]
```diff
@@ -3,8 +3,13 @@ package hazelcast
 import (
 	"context"
 	"fmt"
+	"strings"
 	"time"
 
+	"github.com/hazelcast/hazelcast-platform-operator/controllers/util"
```
Comment (mtyazici) [inline controllers/hazelcast/hazelcast_status.go:9]: Could you order the imports?

File: controllers/hazelcast/hot_backup_controller.go [hunk 3]
```diff
@@ -2,7 +2,13 @@ package hazelcast
 
 import (
 	"context"
+	"encoding/json"
 
+	n "github.com/hazelcast/hazelcast-platform-operator/controllers/naming"
+	"github.com/hazelcast/hazelcast-platform-operator/controllers/util"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
```
Comment (mtyazici) [inline controllers/hazelcast/hot_backup_controller.go:9]: Same as above

**Review by hasancelik — APPROVED** [review]
LGTM 👍 the issue that @mtyazici mentioned should be addressed before the merge.

**Comment by hasancelik** [comment]
Current sample CR statuses after last commit:

```
  Status:
  Hazelcast Cluster Status:
    Ready Members:  0/3
  Members:
    Connected:  false
    Ip:         10.164.2.23
    Message:    back-off 10s restarting failed container=hazelcast pod=hazelcast-0_default(07c1a692-1be9-408d-b245-bb93cf01af66)
    Pod Name:   hazelcast-0
    Reason:     CrashLoopBackOff
  Message:      multiple (1) errors: pod hazelcast-0 in namespace default failed for back-off 10s restarting failed container=hazelcast pod=hazelcast-0_default(07c1a692-1be9-408d-b245-bb93cf01af66):

-----------

Status:
  Hazelcast Cluster Status:
    Ready Members:  3/3
  Members:
    Connected:  true
    Ip:         10.164.2.27
    UID:        85eeb6f3-00a3-4bed-aa4c-30959d6bda67
    Version:    5.0.2
    Connected:  true
    Ip:         10.164.1.12
    UID:        e546754e-209d-4ac5-9fd0-119a6d8bf371
    Version:    5.0.2
    Connected:  true
    Ip:         10.164.0.14
    UID:        5a7b1a1f-4a1c-47ff-8151-24db0e5db51e
    Version:    5.0.2
  Phase:        Running
```

**Review by mtyazici — APPROVED** [review]

**Pull request merged**